database-commons/src/main/java/io/cdap/plugin/util/DBUtils.java

```diff
@@ -258,8 +256,7 @@ public static Object transformValue(int sqlType, int precision, int scale,
           return ((Number) original).intValue();
         case Types.NUMERIC:
         case Types.DECIMAL:
-          BigDecimal decimal = (BigDecimal) original;
-          return new BigDecimal(decimal.unscaledValue(), scale, new MathContext(precision));
+          return original;
```

Comment:
shouldn't we at least try to cast value to make sure it actually returns needed type?

Reply:
Fixed.